ballista/rust/core/src/serde/physical_plan/mod.rs

```diff
@@ -1029,14 +1029,10 @@ fn decode_scan_config(
         .map(|f| f.try_into())
         .collect::<Result<Vec<_>, _>>()?;
 
-    let object_store = if let Some(file) = file_groups.get(0).and_then(|h| h.get(0)) {
-        runtime.object_store(file.file_meta.path())?.0
-    } else {
-        Arc::new(LocalFileSystem {})
-    };
+    // TODO: This will not roundtrip object storage correctly as it loses the scheme
```

Comment:
I'm not really sure how to handle this, as at this point the scheme has been stripped from the paths. I suspect something else needs to be encoded in FileScanExecConf - but I'm not very familiar with this code

Reply:
Yeah, I think originally I had added a `fn scheme(&self) -> String` method to `ObjectStore` and then we can add that into the serialized `FileScanExecConf`. 

I got rid of it before merging because I thought using the scheme from the file URI would be simpler, but I guess we need to bring it back.

Reply:
Yeah, I think it should probably round-trip scheme and host explicitly and use this to determine the object store. Currently ObjectStoreRegistry is a tad confused w.r.t. host, but its something that I intend to fix soon so would be good to cover it as well.